File: src/to/us/harha/parallelray/util/math/Intersection.java
package to.us.harha.parallelray.util.math;

public class Intersection
{

	private Vec3f m_pos;
	private Vec3f m_norm;
	private float m_t;

	public Intersection(Vec3f pos, Vec3f norm, float t)
	{
		m_pos = pos;
		m_norm = norm;
		m_t = t;
	}

	public Intersection()
	{
		m_pos = new Vec3f();
		m_norm = new Vec3f();
		m_t = 0.0f;
	}

	public Vec3f getPos()
	{
		return m_pos;
	}

	public Vec3f getNorm()
	{
		return m_norm;
	}

	public float getT()
	{
		return m_t;
	}

	public void setPos(Vec3f pos)
	{
		m_pos.set(pos);
	}

	public void setNorm(Vec3f norm)
	{
		m_norm.set(norm);
	}

	public void setT(float t)
	{
		m_t = t;
	}

}

File: src/nid/parallelray/util/math/Intersection.ts
import {Vec3f} from "./Vec3f";

export class Intersection
{

	private m_pos:Vec3f;
	private m_norm:Vec3f;
	private m_t:number;

	constructor(pos:Vec3f = new Vec3f(), norm:Vec3f = new Vec3f(), t:number=0.0)
	{
		this.m_pos = pos;
		this.m_norm = norm;
		this.m_t = t;
	}

	getPos():Vec3f
	{
		return this.m_pos;
	}

	getNorm():Vec3f
	{
		return this.m_norm;
	}

	getT():number
	{
		return this.m_t;
	}

	setPos(pos:Vec3f):void
	{
		this.m_pos.set(pos);
	}

	setNorm(norm:Vec3f):void
	{
		this.m_norm.set(norm);
	}

	setT(t:number):void
	{
		this.m_t = t;
	}

}
